marathon-gradle-plugin/src/main/kotlin/com/malinskiy/marathon/MarathonPlugin.kt

```diff
@@ -46,38 +48,36 @@ class MarathonPlugin : Plugin<Project> {
 
         val marathonWorkerTask = rootProject.tasks.named(WORKER_TASK_NAME, MarathonWorkerRunTask::class.java)
         androidComponents.onVariants { variant ->
-            variant.nestedComponents
-                .filterIsInstance<AndroidTest>()
-                .forEach { androidTest ->
-                    val testTaskForVariant = project.registerTaskForTestVariant(variant, androidTest, marathonWorkerTask)
-                    marathonTask.configure { dependsOn(testTaskForVariant) }
+            variant.components
+                .filter { it is GeneratesTestApk }
+                .forEach { component ->
+                    val testTask = registerTestTask(variant, component, marathonWorkerTask)
+                    marathonTask.configure { dependsOn(testTask) }
                 }
         }
     }
 
-    private fun Project.registerTaskForTestVariant(
+    private fun Project.registerTestTask(
         variant: Variant,
-        androidTest: AndroidTest,
+        testComponent: Component,
         marathonWorkerTask: TaskProvider<MarathonWorkerRunTask>
-    ): TaskProvider<MarathonScheduleTestsToWorkerTask> {
-        return tasks.register("$TASK_PREFIX${androidTest.name.capitalized()}", MarathonScheduleTestsToWorkerTask::class.java) {
+    ): TaskProvider<MarathonScheduleTestsToWorkerTask> =
+        tasks.register("$TASK_PREFIX${variant.name.capitalized()}AndroidTest", MarathonScheduleTestsToWorkerTask::class.java) {
```

Comment:
`TestVariant` name doesn't have `AndroidTest` prefix, so we have to hard-code it to have consistent naming for application, library and test modules.